### PR TITLE
add type hinting to events

### PR DESCRIPTION
**Pull request by adhami3310**

### CODE REVIEW

**Comment by adhami3310** [comment]
there's still more `EventHandler[lambda ...]`, more will be coming soon

File: reflex/utils/pyi_generator.py [hunk 1]
```diff
@@ -427,18 +427,59 @@ def _generate_component_create_functiondef(
     all_props = [arg[0].arg for arg in prop_kwargs]
     kwargs.extend(prop_kwargs)
 
+    def figure_out_return_type(annotation: Any):
+        print(annotation)
```
Comment (masenf) [inline reflex/utils/pyi_generator.py:431]: stray print
```suggestion
```

File: reflex/event.py [hunk 2]
```diff
@@ -1020,13 +1055,15 @@ def parse_args_spec(arg_spec: ArgsSpec):
...
     annotations = get_type_hints(arg_spec)
 
-    return arg_spec(
-        *[
-            Var(f"_{l_arg}").to(
-                unwrap_var_annotation(resolve_annotation(annotations, l_arg))
-            )
-            for l_arg in spec.args
-        ]
+    return list(
```
Comment (masenf) [inline reflex/event.py:1058]: why does this return a `list` now? this is the only change i don't really understand.

Comment (adhami3310) [reply reflex/event.py:1058]: because argspec is now typed as "sequence"

**Review by masenf — APPROVED** [review]

**Pull request merged**